wiiu/input/wiiu_hid.c

```diff
@@ -458,16 +467,19 @@ static void wiiu_hid_polling_thread_cleanup(OSThread *thread, void *stack) {
 }
 
 static void wiiu_handle_attach_events(wiiu_hid_t *hid, wiiu_attach_event *list) {
-  wiiu_attach_event *event;
+  wiiu_attach_event *event, *event_next = NULL;
+
   if(!hid || !list)
     return;
 
-  for(event = list; event != NULL; event = event->next) {
+  for(event = list; event != NULL; event = event_next ) {
+    event_next  = event->next;
     if(event->type == HID_DEVICE_ATTACH) {
```

Comment:
Some minor style nits BTW - if a conditional block only has one line, drop the brackets around it.